ProjFS.Mac/PrjFSKextTests/MockVnodeAndMount.hpp

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "../PrjFSKext/kernel-header-wrappers/vnode.h"
+#include "../PrjFSKext/kernel-header-wrappers/mount.h"
+#include "../PrjFSKext/public/FsidInode.h"
+#include <memory>
+#include <string>
+
+typedef std::shared_ptr<mount> MountPointer;
+typedef std::shared_ptr<vnode> VnodePointer;
+typedef std::weak_ptr<vnode> VnodeWeakPointer;
+
+struct mount
+{
+private:
+    vfsstatfs statfs;
+    uint64_t nextInode;
+    
+public:
+    static MountPointer Create(const char* fileSystemTypeName, fsid_t fsid, uint64_t initialInode);
+    
+    inline fsid_t GetFsid() const { return this->statfs.f_fsid; }
+    
+    friend struct vnode;
+    friend vfsstatfs* vfs_statfs(mount_t mountPoint);
+    friend FsidInode Vnode_GetFsidAndInode(vnode_t vnode, vfs_context_t vfsContext);
+};
+
+struct vnode
+{
+private:
+    VnodeWeakPointer weakSelfPointer;
+    MountPointer mountPoint;
+    
+    uint64_t inode;
+    uint32_t vid;
+    int32_t ioCount = 0;
+    bool isRecycling = false;
+    
+    errno_t getPathError = 0;
+    
+    vtype type = VREG;
+    
+    std::string path;
+    const char* name;
+    
+    void SetPath(const std::string& path);
+
+    explicit vnode(const MountPointer& mount);
+    
+    vnode(const vnode&) = delete;
+    vnode& operator=(const vnode&) = delete;
+    
+public:
+    static VnodePointer Create(const MountPointer& mount, const char* path, vtype vnodeType = VREG);
+    static VnodePointer Create(const MountPointer& mount, const char* path, vtype vnodeType, uint64_t inode);
+    ~vnode();
+    
+    uint64_t GetInode() const { return this->inode; }
+    uint32_t GetVid() const { return this->vid; }
+    void SetGetPathError(errno_t error);
+    void StartRecycling();
+
+    friend int vnode_isrecycled(vnode_t vnode);
```

Comment:
Should these be friend?

Reply:
Yeah, that ended up messy. Where it's sensible, I'll just make the fields they access in `struct vnode` public, and where it makes more sense, I'll add some public member functions that those mocked functions can call.